lib/beaker/test_suite.rb

```diff
@@ -172,7 +172,9 @@ def write_junit_xml(xml_file)
             end
             suite.add_child(properties)
 
-            @test_cases.each do |test|
+            test_cases_to_report = @test_cases
```

Comment:
It's not clear to me if this is creating a reference to `@test_cases` or actually creating a copy of that data structure.

Could you add a spec test to validate that the order of test_cases is not affected by the `time_sort` option?